File: src/etch/bindings/java/transport/Resources.java
package etch.bindings.java.transport;

import java.util.HashMap;

/**
 * A HashMap of string to object.
 */
public class Resources extends HashMap<String, Object>
{
	private static final long serialVersionUID = 1875926663269692773L;

	public Resources()
	{
		// nothing to do.
	}
	
	public Resources( Resources r )
	{
		super( r );
	}
}

File: src/etch/bindings/csharp/transport/Resources.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Transport
{
    public class Resources : Dictionary<String, Object>
    {
        public Resources()
        {
            // ignore
        }

        public Resources( Resources r )
            : base( r )
        { }

        public Object Get( String key )
        {
            try
            {
                return this[ key ];
            }
            catch ( KeyNotFoundException )
            {
                return null;
            }
        }
    }
}
